src/syntaxtree.c

```diff
@@ -236,15 +333,66 @@ static Obj SyntaxTreeRecExpr(Obj result, Expr expr)
     return result;
 }
 
-static Obj SyntaxTreeFloatLazy(Obj result, Expr expr)
+static Expr SyntaxTreeCodeRecExpr(Obj node)
 {
-    Obj string;
+    Obj   keyvalue = ELM_REC(node, RNamName("keyvalue"));
+    UInt  len = LEN_LIST(keyvalue);
+    UInt1 tnum = FIND_TNUM_FROM_STRING(ELM_REC(node, RNamName("type")));
+    Expr  record = NewStatWithProf(tnum, 2 * len * sizeof(Expr), 0);
+    for (int i = 0; i < len; i++) {
+        Obj  keyvaluepair = ELM_PLIST(keyvalue, i + 1);
+        Obj  keynode = ELM_REC(keyvaluepair, RNamName("key"));
+        Obj  valuenode = ELM_REC(keyvaluepair, RNamName("value"));
+        Expr key;
+        if (IS_STRING(keynode)) {
+            key = INTEXPR_INT(RNamObj(keynode));
+        }
+        else {
+            key = SyntaxTreeDefaultCoder(keynode);
+        }
+        Expr value = SyntaxTreeDefaultCoder(valuenode);
+        WRITE_EXPR(record, 2 * i, key);
+        WRITE_EXPR(record, 2 * i + 1, value);
+    }
+    return record;
+}
 
-    string = GET_VALUE_FROM_CURRENT_BODY(READ_EXPR(expr, 1));
+static Obj SyntaxTreeFloatLazy(Obj result, Expr expr)
+{
+    Obj string = GET_VALUE_FROM_CURRENT_BODY(READ_EXPR(expr, 1));
     AssPRec(result, RNamName("value"), string);
     return result;
 }
 
+static Obj SyntaxTreeFloatEager(Obj result, Expr expr)
+{
+    Obj  value = GET_VALUE_FROM_CURRENT_BODY(READ_EXPR(expr, 0));
+    Obj  string = GET_VALUE_FROM_CURRENT_BODY(READ_EXPR(expr, 1));
+    UInt char_val = (UInt)READ_EXPR(expr, 2);
```

Comment:
```suggestion
    UChar chr = (UChar)READ_EXPR(expr, 2);
```
Otherwise an out of bounds access could happen a few lines later on.

And `char_val` violates our naming conventions.